### PR TITLE
Remove blowfish from transport ciphers

### LINKED ISSUE #2038: paramiko causing Warnings due to Blowfish deprecation in cryptography 37.0.0
cryptography released a new version 37.0.0 in which they deprecated blowfish: https://github.com/pyca/cryptography/pull/6922

We're now getting this warning output:
```
tests/test_dag_contents.py:7: in <module>
    import rail
.venv/lib/python3.9/site-packages/rail/__init__.py:33: in <module>
    from .operators.network.sftp_any_file_sensor import SFTPAnyFileSensor
.venv/lib/python3.9/site-packages/rail/operators/network/sftp_any_file_sensor.py:5: in <module>
    from airflow.providers.sftp.hooks.sftp import SFTPHook
.venv/lib/python3.9/site-packages/airflow/providers/sftp/hooks/sftp.py:24: in <module>
    import pysftp
.venv/lib/python3.9/site-packages/pysftp/__init__.py:12: in <module>
    import paramiko
.venv/lib/python3.9/site-packages/paramiko/__init__.py:22: in <module>
    from paramiko.transport import SecurityOptions, Transport
.venv/lib/python3.9/site-packages/paramiko/transport.py:132: in <module>
    class Transport(threading.Thread, ClosingContextManager):
.venv/lib/python3.9/site-packages/paramiko/transport.py:236: in Transport
    "class": algorithms.Blowfish,
.venv/lib/python3.9/site-packages/cryptography/utils.py:123: in __getattr__
    warnings.warn(obj.message, obj.warning_class, stacklevel=2)
E   cryptography.utils.CryptographyDeprecationWarning: Blowfish has been deprecated
```

During part of our build as paramiko appears to be referencing blowfish in it's transport options.

I'd be more then willing to submit a PR to remove blowfish from the paramiko transport ciphers to clear this deprecation warning.

### PR DESCRIPTION
**Pull request by repl-mike-roest**
hopefully fixes #2038

### CODE REVIEW

**Comment by kevin-bates** [comment]
If it helps anything, these changes resolve our current [CI issue](https://github.com/jupyter-server/enterprise_gateway/runs/6181669385?check_suite_focus=true#step:9:193).

**Comment by earlruby** [comment]
LGTM

**Review by ajaysuwalka — APPROVED** [review]

**Comment by andreif** [comment]
@bitprophet Please

File: tests/test_transport.py [hunk 1]
```diff
@@ -196,9 +196,9 @@ def setup_test_server(
     def test_security_options(self):
         o = self.tc.get_security_options()
         self.assertEqual(type(o), SecurityOptions)
-        self.assertTrue(("aes256-cbc", "blowfish-cbc") != o.ciphers)
-        o.ciphers = ("aes256-cbc", "blowfish-cbc")
-        self.assertEqual(("aes256-cbc", "blowfish-cbc"), o.ciphers)
+        self.assertTrue(("aes256-cbc", "aes192-cbc") != o.ciphers)
```
Comment (bskinn) [inline tests/test_transport.py:199]: For my info -- why the change from `blowfish-cbc` to `aes192-cbc` here? Would it make sense to add additional ciphers to this test?

Comment (repl-mike-roest) [reply tests/test_transport.py:199]: I just changed it to another valid cipher that was in the list to maintain the functionality of the test where it validates the cipher list can be overridden.  I just picked one at random effectively

Comment (bskinn) [reply tests/test_transport.py:199]: Gotcha, figured that was it. Thanks!

**Comment by bskinn** [comment]
Thanks very much for putting together the PR for this!

From talking to @bitprophet, given that workable pinning/pip-fu strategies have been identified to deal with these warnings, there's not going to be a specific release with these changes. However, they will get pulled into a future release.

**Comment by andreif** [comment]
Do you plan to merge just before the release or it can go to master/main already now?

**Comment by bskinn** [comment]
It'll probably be merged just before the release, @andreif. @bitprophet may want also to backport the changes to earlier release lines, and if so there's a lot of cherry-picking that happens there. Easier logistically to have the PR poised for merge, IIUC, rather than already merged in.

**Comment by bitprophet** [comment]
This is a tricky one because it's arguably removing functionality. I'm not quite so strict as to imagine enough people are using Blowfish that nuking it has to go in a "truly backwards incompatible" major release. But I am on the fence re: whether it can qualify as a bugfix. I think it'd work best as a quick 2.11 release.

**Comment by earlruby** [comment]
Blowfish has been deprecated. and the deprecation caused an error message, so I'd call it a bug fix.
